### PR TITLE
Hyper Server example + move other sample projects to examples/

### PR DESCRIPTION
**Pull request by puhrez**
Works the `hyper_server` spec'd in #29

### CODE REVIEW

**Comment by puhrez** [comment]
@cybergeek94 chuggin' along

**Comment by cybergeek94** [comment]
You shouldn't be having problems with type mismatches unless the `hyper` versions for `multipart` and the sample project aren't exactly the same. I would check that.

I'm thinking of moving sample projects to being just single files under `examples/` that can be run with `cargo run --example`. That's what I'm doing with `multipart-async`, anyway. It's much simpler and you don't have to deal with version mismatch. Wanna try that?

**Comment by puhrez** [comment]
@cybergeek94 this is on a branch?

**Comment by cybergeek94** [comment]
@puhrez I'm sorry, I don't understand. Is what on a branch? I haven't started moving sample projects to [Cargo examples](http://doc.crates.io/manifest.html#examples) yet, I was just wondering if you thought it was a good idea.

The other samples have outdated versions of `multipart` as well. I'll need to fix that manually or move them to being examples so that I don't have to keep up on that.

**Comment by puhrez** [comment]
I meant the analogous work you've done on multipart-async. But yeah, I think moving them to the manifest's standard is good idea.

**Comment by puhrez** [comment]
@cybergeek94 So i migrated things over to examples, unfortunately now since they aren't standalone projects, this kinda breaks my emacs setup, but /shrug, its much easier to build and run them now.

**Comment by cybergeek94** [comment]
@puhrez https://github.com/cybergeek94/multipart-async

It looks like a straight-up copy of this repo because that's how I chose to bootstrap it. The code itself is changed quite a bit, though. I'm still developing it and the `Cargo.toml` is still configured to use local `buf_redux` because I was working on additions to that crate simultaneously.

The usage becomes more clear in [the `examples/` directory](https://github.com/cybergeek94/multipart-async/tree/master/examples). The server-side Hyper usage is a bit clunky because you don't get the request object until after you construct the handler, so you have to do some gymnastics with `Option`. I think I might develop a wrapper around `Option` that derefs directly to the contained value and panicks if it is `None`, to minimize that boilerplate since the handler methods are semi-guaranteed to be called in a consistent order..

File: examples/README.md [hunk 1]
```diff
@@ -22,4 +28,20 @@ Author: [White-Oak](white-oak)
 This sample project shows how to use `multipart` with the [`tiny_http` crate](https://crates.io/crates/tiny_http), via `multipart`'s support for the `tiny_http::Request` type.
 
 
+```
+$ cargo run --features "tiny_http" --example tiny_http
+```
+
+[`hyper_server`](hyper_server)
+-----------------------------
+Author: [Puhrez](puhrez)
```
Comment (cybergeek94) [inline examples/README.md]: This was my mistake earlier in the file, but forward-referenced links actually use two pairs of square brackets like this:

    [Puhrez][puhrez]

With regular parenthesis, it'll try to interpret the link as relative to the current page, which is not what we want.

**Comment by cybergeek94** [comment]
@puhrez Looks good, only nit is in comment on `examples/README.md`.

**Comment by cybergeek94** [comment]
@puhrez Thanks!

**Pull request merged**